compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2548,15 +2356,166 @@ private boolean isIgnoredErrorRefRestVar(BLangErrorVarRef parentErrorVarRef) {
     }
 
     @Override
-    public void visit(BLangAbort abortNode) {
-        BLangReturn returnStmt = ASTBuilderUtil.createReturnStmt(abortNode.pos, symTable.intType, -1L);
-        result = rewrite(returnStmt, env);
+    public void visit(BLangRetry retryNode) {
```

Comment:
Shall we move transaction related code to TransactionDesugar ?

Reply:
is this really transaction related?